### PR TITLE
[GO] Implement language pack endpoints (REST full-CRUD + gRPC get/add)

### LINKED ISSUE #1376: [GO] GRPC LanguagePack
## 📖 Description
Implement AddLanguage and LanguagePackRequest GRPC endpoints
```
message AddLanguagePayload {
  uint32 slot = 1;
  LanguagePack language_pack = 2;
  string language = 3;                  // e.g. "es" Global Spanish.
}

message LanguagePackRequestPayload {
  uint32 slot = 1;     // Uniquely identifies the device at node scope.
  string language = 2; // e.g. "es" Global Spanish.
}
```

---

## 🎯 Goal / Outcome
Language can be added and then queried through GRPC (querying one-of-everything in cpp on port 6254):
AddLanguage:
```
grpcurl -plaintext \
  -import-path ~/Catena/smpte/interface/proto \
  -proto service.proto \
  -d '{
    "slot": 0,
    "language": "nl",
    "languagePack": {
      "name": "Global Dutch",
      "words": {
        "greeting": "Hallo",
        "parting": "Tot ziens"
      }
    }
  }' \
  localhost:6254 st2138.CatenaService/AddLanguage
```
```
{}
```
LanguagePackRequest:
```
grpcurl -plaintext \
  -import-path ~/Catena/smpte/interface/proto \
  -proto service.proto \
  -d '{"slot": 0, "language": "nl"}' \
  localhost:6254 st2138.CatenaService/LanguagePackRequest
```
```
{
  "language": "nl",
  "languagePack": {
    "name": "Global Dutch",
    "words": {
      "greeting": "Hallo",
      "parting": "Tot ziens"
    }
  }
}
```
---

## ✅ Definition of Done (DoD)
A list of TODOs that need to be done for the issue to be done:
- [ ] Feature is implemented and works as expected
- [ ] Edge cases are handled
- [ ] Tests are added or updated (if applicable)
- [ ] Documentation is updated (if needed)
- [ ] Code is reviewed and approved

---

## 🛠️ Implementation Notes (Optional)

---

## 🔗 Related Issues / Links

---
## 🚧 Risks / Open Questions (Optional)

---

### PR DESCRIPTION
**Pull request by ZacharyDuaibesRV**
## 📖 Description
Implements the Go language-pack endpoints for the Catena SDK across **both** the REST and gRPC transports.

A language pack is a display name plus a dictionary of string identifiers (e.g. `"greeting"` → `"Hello"`), all in one language. The language tag (e.g. `"es"`) is carried by the handlers.

**REST** - full CRUD:
- `GET /st2138-api/v1/{slot}/language-pack/{language}` - retrieve a pack
- `POST /st2138-api/v1/{slot}/language-pack/{language}` - add a pack
- `PUT /st2138-api/v1/{slot}/language-pack/{language}` - update a pack
- `DELETE /st2138-api/v1/{slot}/language-pack/{language}` - delete a pack

**gRPC** - get + add (matching the service surface):
- `LanguagePackRequest` - retrieve a pack
- `AddLanguage` - add a pack

> This is a fresh, clean-history replacement for #1375 (closed). It consolidates the gRPC work from #1443 so REST and gRPC land together, since both depend on the same `server.go` changes.

## 🛠️ Implementation Notes
- **`LanguagePack` wrapper**: hides protos behind a fluent builder - exported `Proto`, `WithName`/`WithWord`/`WithWords`, and `GetName`/`GetWords` getters (mirrors `param.go`). Added to the `ResponseType` interface.
- **Handlers**: four separate handlers (get/add/update/delete) wired onto develop's shared `realRegisterHandler`/`invokeHandler`/`realInvokeGate` helpers, with read/write access checks.
- **REST transport**: routes all four verbs; GET builds the outer `DeviceComponent_ComponentLanguagePack`. Fixes GET returning 404 when a handler returns OK with a nil proto.
- **gRPC transport**: implements `LanguagePackRequest` and `AddLanguage`; update/delete stay REST-only (no such RPCs in the proto).
- **Example**: `oneOfEverything` registers all four handlers per slot via a thread-safe `LanguageStore`.
- **Tests**: wrapper, four handlers, REST, and gRPC paths.

## 🔗 Related Issues / Links
- Closes #1376
- References #1130, #1377
- Replaces #1375, supersedes #1443 (gRPC consolidated here)

### CODE REVIEW

File: sdks/go/pkg/catena/server.go [hunk 1]
```diff
@@ -96,6 +97,8 @@ func (e EndpointType) String() string {
 		return "ExecuteCommand"
 	case EndpointParamInfo:
 		return "ParamInfo"
+	case EndpointLanguagePack:
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: should be 4 endpoints

File: sdks/go/pkg/catena/server.go [hunk 2]
```diff
@@ -135,6 +138,11 @@ type ParamInfoHandler func(slot uint16, oidPrefix string, recursive bool, ctx Ha
 // ListLanguagesHandler returns the language codes supported by the device model
 // at a slot (e.g. ["en", "fr"]). An empty slice indicates no multi-lingual support.
 type ListLanguagesHandler func(slot uint16, ctx HandlerContext) ([]string, StatusResult)
+
+type LanguagePackHandler func(slot uint16, language string, ctx HandlerContext) (LanguagePack, StatusResult)
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: decided to use CRUD for the names

File: sdks/go/pkg/catena/server.go [hunk 3]
```diff
@@ -826,6 +866,41 @@ func (s *server) InvokeListLanguagesHandler(slot uint16, transportContext Transp
...
 
+func (s *server) InvokeLanguagePackHandler(slot uint16, language string, transportContext TransportContext) (LanguagePack, StatusResult) {
+	return invokeHandler(s, transportContext, EndpointLanguagePack, false, s.languagePackHandlers, slot,
+		"LanguagePack "+language+" not found at slot "+strconv.Itoa(int(slot)),
+		func(handler LanguagePackHandler, ctx HandlerContext) (LanguagePack, StatusResult) {
+			return handler(slot, language, ctx)
+		})
+}
+
+func (s *server) InvokeAddLanguageHandler(slot uint16, language string, languagePack LanguagePack, transportContext TransportContext) StatusResult {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: spec says everything but `READ` not only requires write access, but also specifically the `adm` scope. so that should be an extra check after you have the HandlerContext

File: sdks/go/pkg/catena/language_pack.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+// WithWords merges the given translation entries into the language pack.
+func (lp LanguagePack) WithWords(words map[string]string) LanguagePack {
+	lp.ensurePack()
+	if lp.Proto.Words == nil {
+		lp.Proto.Words = map[string]string{}
+	}
+	for key, value := range words {
+		lp.Proto.Words[key] = value
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/language_pack.go]: convert to `maps.Copy(lp.Proto.Words, words)`

File: sdks/go/pkg/transports/grpc_transport.go [hunk 5]
```diff
@@ -567,12 +567,73 @@ func ToGRPCCode(s catena.StatusCode) codes.Code {
...
+
+	return &protos.Empty{}, nil
 }
 
 // LanguagePackRequest returns a language pack
 func (s *catenaService) LanguagePackRequest(ctx context.Context, req *protos.LanguagePackRequestPayload) (*protos.DeviceComponent_ComponentLanguagePack, error) {
-	return nil, status.Error(codes.Unimplemented, "LanguagePackRequest not implemented")
+	slot, err := catena.ValidateSlot(req.Slot)
+	if err.Code != catena.StatusCodeOk {
+		return nil, status.Error(ToGRPCCode(err.Code), err.Error)
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/grpc_transport.go:596]: coverage

File: sdks/go/pkg/transports/rest_transport.go [hunk 6]
```diff
@@ -739,6 +743,86 @@ func (t *RestTransport) handleParamInfoEndpoint(w http.ResponseWriter, r *http.R
...
+			return
+		}
+
+		pack := &protos.LanguagePack{}
+		if err := protojson.Unmarshal(body, pack); err != nil {
+			t.writeHTTPStatusResult(w, catena.StatusWithCode(catena.StatusCodeInvalidArgument, "invalid language pack request body"))
+			return
+		}
+
+		languagePack := catena.NewLanguagePack().
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/rest_transport.go]: just do `catena.LanguagePack{Proto: pack}`

File: sdks/go/pkg/transports/rest_transport.go [hunk 7]
```diff
@@ -739,6 +743,86 @@ func (t *RestTransport) handleParamInfoEndpoint(w http.ResponseWriter, r *http.R
 	}
 }
 
+func (t *RestTransport) handleLanguagePackEndpoint(w http.ResponseWriter, r *http.Request, slot uint16, pathParts []string) {
+	if len(pathParts) == 0 || strings.Join(pathParts, "/") == "" {
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/rest_transport.go]: can be simplified to if len(pathPaths) != 1. an empty lang code is for the server/BL to handle

File: sdks/go/pkg/transports/grpc_transport.go [hunk 8]
```diff
@@ -567,12 +567,73 @@ func ToGRPCCode(s catena.StatusCode) codes.Code {
...
+
+	language := req.Language
+	if language == "" {
+		logger.Error("AddLanguage missing language", "slot", slot)
+		return nil, status.Error(codes.InvalidArgument, "language is required")
+	}
+	if req.LanguagePack == nil {
+		logger.Error("AddLanguage missing language pack", "slot", slot, "language", language)
+		return nil, status.Error(codes.InvalidArgument, "language_pack is required")
+	}
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/grpc_transport.go]: These can be checked in the server code instead of in every transport.

File: sdks/go/pkg/transports/grpc_transport.go [hunk 9]
```diff
@@ -567,12 +567,73 @@ func ToGRPCCode(s catena.StatusCode) codes.Code {
...
+
+	logger.Info("AddLanguage", "slot", slot, "language", language)
+
+	pack := catena.NewLanguagePack().
```
Comment (abrown-RV) [inline sdks/go/pkg/transports/grpc_transport.go]: `pack := catena.LanguagePack{ Proto: req }`

this way if we ever add new fields, there's less churn

File: sdks/go/pkg/catena/language_pack.go [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+	return lp
+}
+
+// GetName returns the language pack's display name, or "" if unset.
+func (lp LanguagePack) GetName() string {
+	return lp.Proto.GetName()
+}
+
+// GetWords returns the language pack's word dictionary, or nil if unset.
+func (lp LanguagePack) GetWords() map[string]string {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/language_pack.go:94]: `ensurePack()

**Review by abrown-RV — CHANGES_REQUESTED** [review]
especially for the big _test files. please start using the test/sub test pattern to organize the tests.

File: sdks/go/pkg/catena/server.go (hunk 2 above)
Comment (abrown-RV) [reply sdks/go/pkg/catena/server.go]: Full CRUD please

File: sdks/go/pkg/catena/server.go [hunk 11]
```diff
@@ -826,6 +875,79 @@ func (s *server) InvokeListLanguagesHandler(slot uint16, transportContext Transp
...
+		func(handler LanguagePackHandler, ctx HandlerContext) (LanguagePack, StatusResult) {
+			return handler(slot, language, ctx)
+		})
+}
+
+// requireAdminScope enforces the spec rule that every language-pack operation
+// other than read (create/update/delete) needs the admin scope specifically,
+// not merely any write scope. The coarse write gate in realInvokeGate has
+// already run; this narrows it to adm once the HandlerContext is available.
+func requireAdminScope(ctx HandlerContext) StatusResult {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: I propose HandlerContext.RequireWriteScope(scope string) (and read as well) does the same thing, but more portable. The benefit over just using HasWriteScope is that it generates the correct status code and message boilerplate for you.

BL will probably want this too

File: sdks/go/pkg/catena/server.go [hunk 12]
```diff
@@ -826,6 +875,79 @@ func (s *server) InvokeListLanguagesHandler(slot uint16, transportContext Transp
...
+		return StatusWithCode(StatusCodePermissionDenied, "admin scope required")
+	}
+	return StatusWithCode(StatusCodeOk, "")
+}
+
+func (s *server) InvokeAddLanguageHandler(slot uint16, language string, languagePack LanguagePack, transportContext TransportContext) StatusResult {
+	if language == "" {
+		return StatusWithCode(StatusCodeInvalidArgument, "language is required")
+	}
+	if languagePack.Proto == nil {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: arguement validation goes after token validation, so these should go inside the callback and after the adm scope check. Common wisdom is to not send anything but security errors back if the request is not authenticated.

File: sdks/go/pkg/catena/server.go [hunk 13]
```diff
@@ -105,6 +109,14 @@ func (e EndpointType) String() string {
 		return "ExecuteCommand"
 	case EndpointParamInfo:
 		return "ParamInfo"
+	case EndpointGetLanguagePack:
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: one more CRUD rename
```suggestion
	case EndpointReadLanguagePack:
```

File: sdks/go/pkg/catena/language_pack_test.go [hunk 14]
```diff
@@ -0,0 +1,363 @@
...
+		if lp.GetName() != "French" {
+			t.Errorf("expected name 'French', got %q", lp.GetName())
+		}
+		if lp.GetWords()["greeting"] != "Bonjour" {
+			t.Errorf("expected greeting 'Bonjour', got %q", lp.GetWords()["greeting"])
+		}
+	})
+}
+
+func TestServer_RegisterLanguagePackHandlers(t *testing.T) {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/language_pack_test.go]: TestServer_... goes in server_test.go but nice test/subtest layout!

File: sdks/go/pkg/catena/server.go [hunk 15]
```diff
@@ -920,6 +969,82 @@ func (s *server) InvokeListLanguagesHandler(slot uint16, transportContext Transp
 		})
 }
 
+// requireLanguagePackWriteScope enforces the spec rule that every language-pack
+// operation other than read (create/update/delete) needs the adm scope
+// specifically, not merely any write scope. The coarse write gate in
+// realInvokeGate has already run; this narrows it to adm once the
+// HandlerContext is available, reusing the portable RequireWriteScope helper for
+// the status/message boilerplate.
+func requireLanguagePackWriteScope(ctx HandlerContext) StatusResult {
```
Comment (abrown-RV) [inline sdks/go/pkg/catena/server.go]: unneccessary redirection

**Review by abrown-RV — CHANGES_REQUESTED** [review]

**Review by abrown-RV — APPROVED** [review]
